test/kitchen/test/integration/win-agent-with-cws-option/rspec_datadog/win-agent-with-cws-option_spec.rb

```diff
@@ -0,0 +1,131 @@
+    end
+
+end
+  
+shared_examples_for 'a Windows Agent with CWS running' do
+    # We retrieve the value defined in kitchen.yml because there is no simple way
+    # to set env variables on the target machine or via parameters in Kitchen/Busser
+    # See https://github.com/test-kitchen/test-kitchen/issues/662 for reference
+    let(:expect_cws_installed) {
+        parse_dna().fetch('dd-agent-rspec').fetch('cws_included')
+    }
+    it 'can start security agent' do
+        if expect_cws_installed
+            sa_conf_path = ""
+            sp_conf_path = ""
+            if os != :windows
+                sa_conf_path = "/etc/datadog-agent/security-agent.yaml"
+                sp_conf_path = "/etc/datadog-agent/system-probe.yaml"
+            else
+                sa_conf_path = "#{ENV['ProgramData']}\\Datadog\\security-agent.yaml"
+                sp_conf_path = "#{ENV['ProgramData']}\\Datadog\\system-probe.yaml"
+            end
+            f = File.read(sa_conf_path)
+            confYaml = YAML.load(f)
+            if !confYaml
+                confYaml = {}
+            end
+            if !confYaml.key("runtime_security_config")
+                confYaml["runtime_security_config"] = {}
+            end
+            confYaml["runtime_security_config"]["enabled"] = true
+            File.write(sa_conf_path, confYaml.to_yaml)
+        
+            spf = File.read(sp_conf_path)
+            spconfYaml = YAML.load(spf)
+            if !spconfYaml
+                spconfYaml = {}
+            end
+            if !spconfYaml.key("runtime_security_config")
+                spconfYaml["runtime_security_config"] = {}
+            end
+            spconfYaml["runtime_security_config"]["enabled"] = true
+            File.write(sp_conf_path, spconfYaml.to_yaml)
+        
+            expect(is_service_running?("datadog-system-probe")).to be_falsey
+            expect(is_service_running?("datadog-security-agent")).to be_falsey
+            #restart "datadog-agent"
+            stop "datadog-agent"
+            sleep 10
+            start "datadog-agent"
+            sleep 30
+            expect(is_service_running?("datadogagent")).to be_truthy
+            expect(is_service_running?("datadog-system-probe")).to be_truthy
+            expect(is_service_running?("datadog-security-agent")).to be_truthy
+
+
+            stop "datadog-agent"
+            sleep 30
+            expect(is_service_running?("datadogagent")).to be_falsey
+            expect(is_service_running?("datadog-system-probe")).to be_falsey
+            expect(is_service_running?("datadog-security-agent")).to be_falsey
+        end
+    end
+end
+  
+
+describe 'the agent installed with the cws component' do
+    it_behaves_like 'an installed Agent'
+    it_behaves_like 'a running Agent with no errors'
+    it_behaves_like 'a Windows Agent with CWS driver installed'
+    it_behaves_like 'a Windows Agent with CWS driver disabled'
```

Comment:
Why is this included in this list? the surrounding `it_behaves_like` seem to verify that the cws components are good to go but this one implies that there are still parts that aren't supposed to be running.

Reply:
which one?

Reply:
I _did_ highlight it but Github didn't like it, I guess.

The `a Windows Agent with CWS driver disabled` one.

Reply:
so, the logic based on the way we do drivers
- fresh install (no config changes)
  - agent should be running `it behaves like a running agent no errors
  - driver should be installed `it behaves like a windows agent with cws driver installed`
  - driver should be disabled on initial install; it's set to disabled and the system probe moves it from disabled to manual start on demand `it behaves liek a windows agent with cws driver disabled`
  
  So the test at this point is validating that the result of an install w/o changing configs has all the pieces in the expected state.